### PR TITLE
fix autodoc by adding module as a system path using conf.py

### PR DESCRIPTION
**Pull request by Dalordish**

### CODE REVIEW

**Comment by glipR** [comment]
Hmm, we need the packages installed to do autodoc, unless we wanna chuck imports inside the class definitions. But installing pygame gives errors on the github action. Thoughts @Dalordish ?

**Comment by Dalordish** [comment]
Ah, donk - I tested the process on my local system, which had ev3sim installed and configured properly, but obviously the action isn't setup for that - I'll make some changes

**Comment by Dalordish** [comment]
Have setup CI to install from req.txt as well as install the standard pygame dependencies - let's see if these tests fail

**Comment by glipR** [comment]
> 
> 
> Have setup CI to install from req.txt as well as install the standard pygame dependencies - let's see if these tests fail

Name seems like the github action we use to generate the docs doesn't use the CI python thing you spin up, it creates its own thing.

**Comment by Dalordish** [comment]
> Name seems like the github action we use to generate the docs doesn't use the CI python thing you spin up, it creates its own thing.

Yup, you're right, good catch
I'll probably just manually do the build instead of using the pre-built action

File: .github/workflows/docs_deploy.yml [hunk 1]
```diff
@@ -22,12 +22,14 @@ jobs:
...
+    
     - name: Install dependencies
       run: |
         sudo apt-get update
         sudo apt-get install python3-dev libsdl-image1.2-dev libsdl-mixer1.2-dev libsdl-ttf2.0-dev   libsdl1.2-dev libsmpeg-dev python-numpy subversion libportmidi-dev ffmpeg libswscale-dev libavformat-dev libavcodec-dev
         python -m pip install --upgrade pip
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+        
     - name: Build docs
       working-drectory: ./docs
```
Comment (glipR) [inline .github/workflows/docs_deploy.yml:34]: misspellling here causing build fail

File: .github/workflows/docs_check.yml [hunk 2]
```diff
@@ -22,16 +22,19 @@ jobs:
...
+
     - name: Install dependencies
       run: |
         sudo apt-get update
         sudo apt-get install python3-dev libsdl-image1.2-dev libsdl-mixer1.2-dev libsdl-ttf2.0-dev   libsdl1.2-dev libsmpeg-dev python-numpy subversion libportmidi-dev ffmpeg libswscale-dev libavformat-dev libavcodec-dev
         python -m pip install --upgrade pip
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+
     - name: Build docs
       working-drectory: ./docs
```
Comment (glipR) [inline .github/workflows/docs_check.yml]: same here

File: .github/workflows/docs_deploy.yml (hunk 1 above)
Comment (Dalordish) [reply .github/workflows/docs_deploy.yml]: Fixed, thanks!

**Comment by Dalordish** [comment]
Eureka! The docs artefact seems to include the expected autodoc content now :) 
Have a check over and if it's good then we can merge @glipR

**Comment by glipR** [comment]
You should do the same requirements install in the deploy workflow @Dalordish . Happy to approve after.

**Comment by Dalordish** [comment]
> You should do the same requirements install in the deploy workflow @Dalordish . Happy to approve after.

I was really confused for a moment because I swear I made the change in both - I forgot to save the file 🤦

**Comment by glipR** [comment]
Passing 10 checks! Must be good stuff then ;)

**Pull request merged**